From e8ac633cf9acdd102adb8e4977e009e988670848 Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Thu, 6 Aug 2026 08:27:11 +0200
Subject: [PATCH] cephadm: do not embed random build path into binary

build.py byte-compiles the sources in a directory made by
tempfile.mkdtemp and the random path was recorded in the co_filename
of the code objects shipped in the cephadm zipapp, so every build
produced a different binary.

Pass ddir="" to compileall.compile_dir so that only the path relative
to the build directory is recorded in the pyc files.

This patch was done while working on reproducible builds for openSUSE.
Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1249586

Signed-off-by: Bernhard M. Wiedemann <bwiedemann@suse.de>
---
 src/cephadm/build.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cephadm/build.py b/src/cephadm/build.py
index 4264b814f1..6f359ec560 100755
--- a/src/cephadm/build.py
+++ b/src/cephadm/build.py
@@ -86,6 +86,8 @@ def _compile(dest, tempdir):
         legacy=True,
         quiet=1,
         workers=0,
+        # do not embed the random tempdir path in the pyc files
+        ddir="",
     )
     # TODO we could explicitly pass a python version here
     log.info("Constructing the zipapp file")
